core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3236,16 +3246,23 @@ public void activeSyncMetadata(AlluxioURI path, Collection<AlluxioURI> changedFi
       if (changedFiles == null) {
         // full sync
         LockingScheme lockingScheme = new LockingScheme(path, LockPattern.READ, true);
+        // Populate the status cache before taking a write lock
+        UfsStatusCache statusCache;
+        try (LockedInodePath inodePath =
+            mInodeTree.lockInodePath(lockingScheme.getPath(), LockPattern.READ)) {
```

Comment:
What would be the implications of not even doing any read locking and just calling `populateStatusCache`? I'm not sure what the read lock is providing.

Reply:
The read lock would prevent a mount from being removed before we resolve the mount path